recipes/ncurses/all/conanfile.py

```diff
@@ -0,0 +1,203 @@
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from contextlib import contextmanager
+import os
+
+
+class NCursesConan(ConanFile):
+    name = "ncurses"
+    description = "The ncurses (new curses) library is a free software emulation of curses in System V Release 4.0 (SVr4), and more"
+    topics = ("conan", "ncurses", "terminal", "screen", "tui")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnu.org/software/ncurses"
+    license = "X11"
+    exports_sources = "patches/**"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "pkg_config"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cxx": [True, False],
+        "with_pcre2": [True, False],
+        "with_reentrant": [True, False],
+        "with_widec": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cxx": True,
+        "with_pcre2": False,
+        "with_reentrant": False,
+        "with_widec": False,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.compiler == "Visual Studio":
+            if self.options.with_widec:
+                raise ConanInvalidConfiguration("with_widec is unsupported for Visual Studio")
+            raise ConanInvalidConfiguration("Unsupported on Visual Studio")
+        if self.options.shared:
+            del self.options.fPIC
+        if not self.options.with_cxx:
+            del self.settings.compiler.libcxx
+            del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_pcre2:
+            self.requires("pcre2/10.33")
+
+    def build_requirements(self):
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("getopt-for-visual-studio/20200201")
```

Comment:
this seems to be never actually executed for the reason above (`raise ConanInvalidConfiguration("Unsupported on Visual Studio")`)

Reply:
Same reason as above.
I would prefer to keep it as it might be useful later.
Granted it is dead code.

Reply:
Useful how? Will be visual studio supported in the future?

Reply:
I hope so. I need some feedback/hints from @ThomasDickey .

Reply:
Nice!